File: hello.py
"""A simple hello world module."""


def greet(name: str = "World") -> str:
    """Return a greeting string for the given name.

    Args:
        name: The name to greet. Defaults to "World".

    Returns:
        A greeting string of the form "Hello, <name>!".
    """
    return f"Hello, {name}!"


def main() -> None:
    """Print the default hello world message."""
    print(greet())


if __name__ == "__main__":
    main()

File: hello.js
/**
 * Simple Hello World script.
 * Exports a greeting function and prints it when run directly.
 */

/**
 * Returns a greeting string for the given name.
 * @param {string} [name="World"] - The name to greet.
 * @returns {string} The greeting message.
 */
function greet(name = "World") {
  return `Hello, ${name}!`;
}

// Print the greeting when this file is executed directly
if (require.main === module) {
  console.log(greet());
}

module.exports = { greet };
